corehq/apps/accounting/async_handlers.py

```diff
@@ -212,9 +213,29 @@ def account_response(self):
     @property
     def plan_version_response(self):
         edition = self.data.get('additionalData[edition]')
-        plan_versions = SoftwarePlanVersion.objects.filter(
-            plan__edition=edition
-        )
+        visibility = self.data.get('additionalData[visibility]')
+        most_recent_version = self.data.get('additionalData[most_recent_version]')
+        if most_recent_version == 'True':
+            latest_versions_dates = SoftwarePlanVersion.objects.filter(
+                plan=OuterRef('pk'),
+                plan__edition=edition,
+                plan__visibility=visibility
+            ).order_by('-date_created').values('date_created')[:1]
+            latest_versions = SoftwarePlan.objects.filter(
+                edition=edition,
+                visibility=visibility
```

Comment:
Hi Matt, I agree "this code is more complex if it has an external dependency on the form requiring a value"
Because after I put this on staging. QA flagged that, they're not able to find SoftwarePlan when create new subscription.
My first thought is to add those field in NewSubscriptionForm. 
Do you know how can we ensure that these variables have values? Or can we make edition and visibility optional in our function, so if no edition and no visibility is provided, we return all?